test/lib/absinthe_test.exs

```diff
@@ -200,6 +200,19 @@ defmodule AbsintheTest do
     assert_result {:ok, %{data: %{"thing" => %{"name" => "Foo", "value" => 100}}}}, result
   end
 
+  it "enforces non_null fields in input passed as variable" do
+    query = """
+    mutation UpdateStuff($input: InputStuff!) {
```

Comment:
No need for an issue. Let's go with a query just to make it clear that it isn't something mutation specific.